tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_no_overwrite.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include "testing_common_defs.h"
+#include <addr.h>
+#include <keys.h>
+#include "mocks/hci_core_expects.h"
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - CONFIG_BT_KEYS_OVERWRITE_OLDEST isn't enabled
+ *
+ *  Expected behaviour:
+ *   - NULL reference pointer is returned
+ */
+void test_adding_new_pair_to_full_list(void)
+{
+	struct bt_keys *returned_key;
+	uint8_t id = BT_ADDR_ID_3;
+	bt_addr_le_t *addr = BT_ADDR_LE_3;
+
+	returned_key = bt_keys_get_addr(id, addr);
+
+	expect_not_called_bt_unpair();
+
+	zassert_true(returned_key == NULL, "bt_keys_get_addr() returned a non-NULL reference");
+}
+
+static bool pragma_overwrite_not_enabled(const void *s)
+{
+	if ((IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST))) {
+		return false;
+	}
+
+	return true;
+}
+
+ztest_register_test_suite(bt_keys_get_addr_full_list_no_overwrite, pragma_overwrite_not_enabled,
```

Comment:
it looks like this is deprecated (it is defined in `ztest_test_deprecated.h`). 
Could you update your ztest api calls so they don't use deprecated macros/functions?
For this one, I think the new one is `ZTEST_SUITE`.

Reply:
Yes, was afraid to catching this one, it's going to take some effort :).
I already started using the new APIs with next modules

Reply:
The point is that using the new APIs wasn't available before Torsten's changes, and I've started working on the `Keys` module while the first PR was in review.

So, I'm not sure if I should apply these review comments to the `buf` tests as well or not, please advise?

Reply:
yes could you do that? the PR should be straightforward to review as it's just an API change.

Reply:
> yes could you do that? the PR should be straightforward to review as it's just an API change.

Unfortunately, it doesn't work in that way.

In order to use new APIs, you will need to add `CONFIG_ZTEST_NEW_API=y` to `prj.conf` and once you do that, all deprecated definitions won't be available and you need to switch all APIs to the new ones.

Reply:
My point is that it's doable, but it takes time because the new ones and the deprecated ones aren't identical in terms of suites and tests setup functions and signatures.

I'm currently testing how much effort is required to switch to the new ones and I will starting working on the `buf` module after applying the review comments here to the other commits.